sycl/include/sycl/queue.hpp

```diff
@@ -2784,6 +2784,112 @@ class __SYCL_EXPORT queue : public detail::OwnerLessBase<queue> {
         CodeLoc);
   }
 
+  // no_handler
+
+private:
+  // NOTE: the name of this function - "kernel_single_task" - is used by the
```

Comment:
Can this live outside `queue` directly in `sycl::detail`? If private `queue` access is an issue, then maybe

```
struct LaunchUtils {
  static blah-blah(...) {}
};

class queue {
friend struct LaunchUtils;
}
```